src/components/visiumQC/QPcrResults.tsx

```diff
@@ -0,0 +1,151 @@
+import Panel from '../Panel';
+import React from 'react';
+import Labware from '../labware/Labware';
+import { isSlotFilled } from '../../lib/helpers/slotHelper';
+import RemoveButton from '../buttons/RemoveButton';
+import SlotMeasurements, { MeasurementConfigProps, SlotMeasurement } from '../slotMeasurement/SlotMeasurements';
+import { useFormikContext } from 'formik';
+import { VisiumQCFormData } from '../../pages/VisiumQC';
+import { CDNAProps } from './Amplification';
+
+const QPcrResults = ({ labware, slotMeasurements, removeLabware }: CDNAProps) => {
+  const { values, setErrors, setTouched, setFieldValue } = useFormikContext<VisiumQCFormData>();
+
+  const memoMeasurementConfig: MeasurementConfigProps[] = React.useMemo(
+    () => [
+      {
+        name: 'Cq value',
+        stepIncrement: '.01',
+        validateFunction: validateCqMeasurementValue,
+        initialMeasurementVal: ''
+      }
+    ],
+    []
+  );
+
+  /***
+   * When labwares changes, the slotMeasurements has to be initialized accordingly
+   */
+  React.useEffect(() => {
+    setErrors({});
+    setTouched({});
+
+    if (!labware) {
+      return;
+    }
+    setFieldValue('barcode', labware.barcode);
+    const slotMeasurements: SlotMeasurement[] = labware.slots.filter(isSlotFilled).flatMap((slot) => {
+      return memoMeasurementConfig.map((measurement) => {
+        return {
+          address: slot.address,
+          name: measurement.name,
+          value: measurement.initialMeasurementVal,
+          samples: slot.samples
+        };
+      });
+    });
+    setFieldValue('slotMeasurements', slotMeasurements);
+  }, [labware, setErrors, setTouched, setFieldValue, memoMeasurementConfig]);
+
+  const handleChangeMeasurement = React.useCallback(
+    (measurementName: string, measurementValue: string) => {
+      setFieldValue(measurementName, measurementValue, true);
+    },
+    [setFieldValue]
+  );
+
+  const handleChangeAllMeasurements = React.useCallback(
+    (measurementName: string, measurementValue: string) => {
+      //Reset Errors
+      setErrors({});
+      setTouched({});
+      const measurements = values?.slotMeasurements ? [...values.slotMeasurements] : [];
+      measurements
+        ?.filter((measurement) => measurement.name === measurementName)
```

Comment:
Why we need to filter it? I thougt we only have one measurement (Cq value) here? or Is it for future additions?